extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOIDCProvidersRecorder.java

```diff
@@ -0,0 +1,111 @@
+package io.quarkus.oidc.runtime.providers;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.quarkus.oidc.OidcTenantConfig;
+import io.quarkus.oidc.runtime.OidcConfig;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class KnownOIDCProvidersRecorder {
+
+    final OidcConfig config;
+
+    public KnownOIDCProvidersRecorder(OidcConfig config) {
+        this.config = config;
+    }
+
+    public Supplier<Optional<OidcTenantConfig>> github() {
+        return new Supplier<Optional<OidcTenantConfig>>() {
+            @Override
+            public Optional<OidcTenantConfig> get() {
+                if (config.provider.github.isEmpty()) {
+                    return Optional.empty();
+                }
+                OidcTenantConfig ret = new OidcTenantConfig();
+                ret.clientId = Optional.of(config.provider.github.get().clientId);
+                ret.credentials.secret = Optional.of(config.provider.github.get().secret);
+                ret.authServerUrl = Optional.of("https://github.com/login/oauth");
+                //TODO: do we want to hard code this?
+                ret.applicationType = OidcTenantConfig.ApplicationType.HYBRID;
```

Comment:
`HYBRID` won't work for GitHub - `quarkus-oidc` `service` apps can not handle opaque tokens from providers like GitHub since it has no introspection endpoint - I've opened an issue to make sure a userinfo check is enough